src/oss/confluent.sh

```diff
@@ -978,6 +979,107 @@ print_current() {
+      return 1
+    else
+      return 0
+    fi
+
+}
+
+demo_command() {
+    local subcommand="${1}"
+    local demo_name="${2}"
+    local repo="quickstart-demos"
+
+    if [[ $subcommand == "" || $subcommand == "help" ]]; then
+      demo_usage
+      return
+    fi
+
+    # Check network connectivity
+    local network_status=0
+    if ping -q -c 1 -W 1 github.com >/dev/null; then
+      network_status=1
+    else
+      network_status=0
+    fi
+  
+    # Clone the repo
+    if [ ! -d ${confluent_home}/$repo ]; then
+      if [ $network_status == 1 ]; then
+        git clone https://github.com/confluentinc/$repo.git ${confluent_home}/$repo
+        status=$?
+        if [[ ${status} -ne 0 ]]; then
+            echo "'git clone https://github.com/confluentinc/$repo.git ${confluent_home}/$repo' failed. Please check your git access"
+            return
+        fi
+        cp_version=$(confluent version | awk -F':' '{print $2;}' | awk '$1 > 0 { print substr($1,1,3)}')
+        cd ${confluent_home}/$repo && git checkout $cp_version 2> /dev/null
+      else
+        echo "'confluent demo $subcommand $demo_name' requires network connectivity. Please try again when you are connected."
+        return
+      fi
+    fi
+
+    cd ${confluent_home}/$repo
+    if [[ ${status} -ne 0 ]]; then
+        echo "Cannot find ${confluent_home}/$repo.  Exiting"
+        return
+    fi
+
+    cp_version=$(confluent version | awk -F':' '{print $2;}' | awk '$1 > 0 { print substr($1,1,3)}')
+    if [[ $subcommand == "list" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      echo -e "Available demos from ${confluent_home}/$repo (start a demo 'confluent demo start <demo-name>':"
+      ls | grep -v "README.md" | grep -v "utils" | grep -v "LICENSE"
+    elif [[ $subcommand == "refresh" ]]; then
+      if [ $network_status == 1 ]; then
+        cd ${confluent_home}/$repo && git fetch --tags ; git checkout $cp_version
+      else
+        echo "Running 'confluent demo $subcommand $demo_name' requires network connectivity. Please try again when you are connected."
+        return
+      fi
+    elif [[ $subcommand == "start" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      is_demo_name_valid "$demo_name" "$repo" "$subcommand" || die
+      cd $demo_name
+      ./start.sh
+    elif [[ $subcommand == "stop" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      is_demo_name_valid "$demo_name" "$repo" "$subcommand" || die
+      cd $demo_name
+      ./stop.sh
+    elif [[ $subcommand == "info" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      is_demo_name_valid "$demo_name" "$repo" "$subcommand" || die
+      echo "Please see https://github.com/confluentinc/quickstart-demos/blob/${cp_version}/${demo_name}/README.md"
```

Comment:
This command always prints the url for the README on github and thus seems to have little value. Would it make sense to print part of the contents of the README that's already available locally? (not sure myself, but still just printing the url is not too useful).

Reply:
Agree.